docs(queries): fix typo in view queries documentation

alirezamirehe opened the pull request:
docs(queries): fix typo and clarify view queries documentation

The typo in the [queries document](https://angular.dev/guide/components/queries#content-queries) has been corrected:
Queries never **pierce** through component boundaries.

Updated the documentation to clarify how view queries work.

BREAKING CHANGE: None

alirezamirehe committed:
Update queries.md

The typo in the queries document:

Queries never pierce through component boundaries.

JeanMeche commented:
Hi, could you please [sign the CLA](https://github.com/angular/angular/pull/57146/checks?check_run_id=27971861730) and reword your commit. (You only changed the PR title). 
The commit message should be in the form of `docs: fix typo in view queries documentation`, without a scope.

alirezamirehe committed:
docs: fix typo in view queries documentation

The typo in the [queries document](https://angular.dev/guide/components/queries#content-queries) has been corrected:
Queries never **pierce** through component boundaries.

BREAKING CHANGE: None

JeanMeche commented:
Could you please squash both commits, we'd like this change to be a single commit. Thank you.

alirezamirehe committed:
docs(queries): fix typo in view queries documentation

The typo in the [queries document](https://angular.dev/guide/components/queries#content-queries) has been corrected:
Queries never **pierce** through component boundaries.

BREAKING CHANGE: None

JeanMeche commented:
This PR is still made out of 3 individual commits, we'd need you to squash them in a single one.
